neo/Network/P2P/TaskManager.cs

```diff
@@ -272,5 +277,13 @@ internal protected override bool IsHighPriority(object message)
                     return false;
             }
         }
+
+        internal protected override bool ShallDrop(object message, IEnumerable queue)
+        {
+            if (!(message is TaskManager.NewTasks tasks)) return false;
```

Comment:
@yongjiema, maybe add this case before the removal of duplicated tasks on line 285:
check if `message` is a task with `tasks.Payload.Type == InventoryType.Consensus`.

Because I believe that we do not need the next check for `blocks`, correct?

Reply:
Do you mean adding the below code between line 283 and line 285?

```cs
if (tasks.Payload.Type == InventoryType.Consensus) return false;
```

But I think it doesn't matter for `Consensus` or `Block` as both are high priority, I think you can try `TX`, but I think it may slow down the transaction broadcast.

Reply:
I mean the contrary `if (tasks.Payload.Type != InventoryType.Consensus) return false;`
aehauheuaea

Reply:
Can try it as most duplicated tasks are consensus, but it still got a chance to make the sync stuck or the empty block happen.

Reply:
@yongjiema, what I am saying is exactly in this line of reasoning, to avoid the check for blocks, since the most duplicated are consensus. Thus, we would save some efficiency be not needing to check the Queue when it is not Consensus.